openstudiocore/src/model/Model_Impl.hpp

```diff
@@ -312,6 +312,29 @@ namespace detail {
     void clearCachedYearDescription(const Handle& handle);
     void clearCachedWeatherFile(const Handle& handle);
 
+    typedef std::function<std::shared_ptr<openstudio::detail::WorkspaceObject_Impl>(Model_Impl *, const std::shared_ptr<openstudio::detail::WorkspaceObject_Impl>&, bool)> CopyConstructorFunction;
+    typedef std::map<IddObjectType, CopyConstructorFunction> CopyConstructorMap;
```

Comment:
I would consider using `unordered_map`. Since it'll have faster lookup, especially if you use an `int` key.

Reply:
@kbenne does all this stuff need to be in the Model_Impl header?  Can it just be in the cpp file?  It's not anything we want exported.

Reply:
I guess it is private but still wondering

Reply:
I'm all for @mbadams5 's ideas if they are faster